bundles/org.openhab.binding.tuya/src/main/java/org/openhab/binding/tuya/internal/handler/TuyaDeviceHandler.java

```diff
@@ -508,6 +546,28 @@ private void addChannels(Map<String, SchemaDp> schema) {
                 channeltypeUID = CHANNEL_TYPE_UID_NUMBER;
                 configuration.put("min", schemaDp.min);
                 configuration.put("max", schemaDp.max);
+
+                if (schemaDp.scale > 0 || !schemaDp.unit.isEmpty()) {
+                    channeltypeUID = CHANNEL_TYPE_UID_QUANTITY;
+
+                    if (!schemaDp.unit.isEmpty()) {
+                        Unit<?> unit = schemaDp.parsedUnit;
+                        if (unit == null) {
```

Comment:
Why would unit ever _not_ be null at this point? AFAICT this method should only ever be called once per channel

Reply:
Schemas describe a product type and you can have multiple devices of that type. This code is called when the handler creates Channels for the Thing based on the schema. So it's only called when the handler initializes and some other handler might already have done the parsing.